src/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -438,15 +476,14 @@ private static int FromWin32Value(int value)
         }
 #endif
 
-#if FEATURE_SYSTEM_EVENTS
-        private static void OnUserPreferenceChanging(object sender, UserPreferenceChangingEventArgs e)
+        private static void OnUserPreferenceChanging(object sender, EventArgs args)
         {
-            if (e.Category == UserPreferenceCategory.Color && s_colorTable != null)
+            // Invoking the Delegate.
+            if (s_categoryGetter != null && s_categoryGetter(args) == 2 && s_colorTable != null) // UserPreferenceCategory.Color = 2
```

Comment:
Would it make sense to check s_colorTable != null before invoking s_categoryGetter?